src/sentry/search/events/builder/spans_metrics.py

```diff
@@ -42,9 +52,64 @@ def resolve_metric_index(self, value: str) -> Optional[int]:
 
         return super().resolve_metric_index(value)
 
+    def resolve_split_granularity(self) -> Tuple[List[Condition], Optional[Granularity]]:
+        if self.end is None or self.start is None:
+            raise ValueError("skip_time_conditions must be False when calling this method")
+
+        # Only split granularity when granularity is 1h or 1m
+        # This is cause if its 1d we're already as efficient as possible, but we could add 1d in the future if there are
+        # accuracy issues
+        if self.granularity.granularity == 86400:
+            return [], self.granularity
+        granularity = self.granularity.granularity
+        self.granularity = None
+
+        if granularity == 60:
+            rounding_function = remove_minutes
+            base_granularity = constants.METRICS_GRANULARITY_MAPPING[60]
+        elif granularity == 3600:
+            rounding_function = remove_hours
+            base_granularity = constants.METRICS_GRANULARITY_MAPPING[3600]
+
+        if rounding_function(self.start, False) > rounding_function(self.end):
+            return [], Granularity(granularity)
+        timestamp = self.column("timestamp")
+        granularity = Column("granularity")
+        return [
+            Or(
+                [
+                    And(
+                        [
+                            Or(
+                                [
+                                    Condition(timestamp, Op.GTE, rounding_function(self.end)),
+                                    Condition(
+                                        timestamp, Op.LT, rounding_function(self.start, False)
+                                    ),
+                                ]
+                            ),
+                            Condition(granularity, Op.EQ, base_granularity),
+                        ]
+                    ),
+                    And(
+                        [
+                            Condition(timestamp, Op.GTE, rounding_function(self.start, False)),
+                            # This op is LT not LTE, here's an example why; a query is from 11:45 to 15:45
+                            # if an event happened at 15:02, its caught by the above condition in the 1min bucket at
+                            # 15:02, but its also caught at the 1hr bucket at 15:00
+                            Condition(timestamp, Op.LT, rounding_function(self.end)),
+                            Condition(granularity, Op.EQ, base_granularity + 1),
+                        ]
+                    ),
```

Comment:
and this part is grabbing buckets that are inside the "rounded" times and at the lower fidelity granularity

Reply:
Can we document this logic as docstrings so we don't have to parse this condition anytime we look at this code 😛

Reply:
Also, will we ever want to do a base_granularity + 2? like will there be cases we base_granularity is a minute but we're querying days worth of data?